### ISSUE TITLE
Error After Download

### ISSUE THREAD

**Issue by Harison117** [Franiac/TwitchLeecher#563]
Hi, Using 1.8.2 twitchleecher and I'm getting this error after finishing download with converting enable
Download task has been started!

TWITCH LEECHER INFO
--------------------------------------------------------------------------------------------
Version: 1.8.2

VOD INFO
--------------------------------------------------------------------------------------------
VOD ID: 891965995
Selected Quality: 1920x1080@30fps (Source)
Download Url: https://www.twitch.tv/videos/891965995
Crop Start: Yes (00:58:35)
Crop End: Yes (01:01:35)

OUTPUT INFO
--------------------------------------------------------------------------------------------
Disable Conversion: No
Output File: D:\Videos\Captures\20210128_891965995_Music.mp4
FFMPEG Path: D:\Programas\TwitchLeecher-master\Twitch Leecher\ffmpeg.exe
Temporary Download Folder: C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be

ACCESS INFO
--------------------------------------------------------------------------------------------
Token: %7B%22authorization%22%3A%7B%22forbidden%22%3Afalse%2C%22reason%22%3A%22%22%7D%2C%22chansub%22%3A%7B%22restricted_bitrates%22%3A%5B%5D%7D%2C%22device_id%22%3Anull%2C%22expires%22%3A1612085598%2C%22https_required%22%3Atrue%2C%22privileged%22%3Afalse%2C%22user_id%22%3Anull%2C%22version%22%3A2%2C%22vod_id%22%3A891965995%7D
Signature: 405ed73afaa41c233bce75cc54a254fe1bf3e9bb
Sub-Only: No
Privileged: No

Creating temporary download directory 'C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be'... done!

Retrieving m3u8 playlist urls for all VOD qualities... done!
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/chunked/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/720p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/480p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/audio_only/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/360p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/160p30/index-dvr.m3u8

Playlist url for selected quality 1920x1080@30fps (Source) is https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/chunked/index-dvr.m3u8

Retrieving playlist... done!
Parsing playlist... done!
Number of video chunks: 1213

Starting parallel video chunk download
Number of video chunks to download: 19
Maximum connection count: 10

Parallel video chunk download is running...

Download of all video chunks complete!

Merging all VOD parts into 'C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be\20210128_891965995_Music.ts'...

Executing 'D:\Programas\TwitchLeecher-master\Twitch Leecher\ffmpeg.exe' on 'C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be\20210128_891965995_Music.ts'...
Command line arguments: -y -ss 7.004 -i "C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be\20210128_891965995_Music.ts" -analyzeduration 2147483647 -probesize 2147483647 -c:v copy -c:a copy -t 180 "D:\Videos\Captures\20210128_891965995_Music.mp4"

ffmpeg version N-93529-g8e3b01e20e Copyright (c) 2000-2019 the FFmpeg developers
built with gcc 8.2.1 (GCC) 20190212
configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
libavutil      56. 26.100 / 56. 26.100
libavcodec     58. 48.101 / 58. 48.101
libavformat    58. 26.101 / 58. 26.101
libavdevice    58.  7.100 / 58.  7.100
libavfilter     7. 48.100 /  7. 48.100
libswscale      5.  4.100 /  5.  4.100
libswresample   3.  4.100 /  3.  4.100
libpostproc    55.  4.100 / 55.  4.100
Input #0, mpegts, from 'C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be\20210128_891965995_Music.ts':
Duration: 00:03:10.00, start: 3575.263000, bitrate: 4502 kb/s
Program 1
Stream #0:0[0x100]: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp, 185 kb/s
Stream #0:1[0x101]: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(progressive), 1920x1080 [SAR 1:1 DAR 16:9], 30 fps, 30 tbr, 90k tbn, 60 tbc
Stream #0:2[0x102]: Data: timed_id3 (ID3  / 0x20334449)
D:\Videos\Captures\20210128_891965995_Music.mp4: No such file or directory

Starting temporary download folder cleanup!
Deleting directory 'C:\Users\haris\AppData\Local\Temp\TL_3c7ea7d5-6e50-4234-8fe8-1c006aa347be'... done!

Download task ended with an error!

System.AggregateException: One or more errors occurred. ---> System.ApplicationException: An error occured while converting the video!
   at TwitchLeecher.Services.Services.ProcessingService.ConvertVideo(Action`1 log, Action`1 setStatus, Action`1 setProgress, Action`1 setIsIndeterminate, String sourceFile, String outputFile, CropInfo cropInfo)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass51_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ApplicationException: An error occured while converting the video!
   at TwitchLeecher.Services.Services.ProcessingService.ConvertVideo(Action`1 log, Action`1 setStatus, Action`1 setProgress, Action`1 setIsIndeterminate, String sourceFile, String outputFile, CropInfo cropInfo)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass51_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()<---

And if I try to disable and download the whole video, this error appears

Download task has been started!

TWITCH LEECHER INFO
--------------------------------------------------------------------------------------------
Version: 1.8.2

VOD INFO
--------------------------------------------------------------------------------------------
VOD ID: 891965995
Selected Quality: 1920x1080@30fps (Source)
Download Url: https://www.twitch.tv/videos/891965995
Crop Start: No
Crop End: No

OUTPUT INFO
--------------------------------------------------------------------------------------------
Disable Conversion: Yes
Output File: D:\Videos\Captures\20210128_891965995_Music.ts
FFMPEG Path: D:\Programas\TwitchLeecher-master\Twitch Leecher\ffmpeg.exe
Temporary Download Folder: C:\Users\haris\AppData\Local\Temp\TL_5fa863ae-791f-4777-b72c-e4e1eb2880a8

ACCESS INFO
--------------------------------------------------------------------------------------------
Token: %7B%22authorization%22%3A%7B%22forbidden%22%3Afalse%2C%22reason%22%3A%22%22%7D%2C%22chansub%22%3A%7B%22restricted_bitrates%22%3A%5B%5D%7D%2C%22device_id%22%3Anull%2C%22expires%22%3A1612083621%2C%22https_required%22%3Atrue%2C%22privileged%22%3Afalse%2C%22user_id%22%3Anull%2C%22version%22%3A2%2C%22vod_id%22%3A891965995%7D
Signature: ccbcc5dd068fbecb271d24ba867a332557092261
Sub-Only: No
Privileged: No

Creating temporary download directory 'C:\Users\haris\AppData\Local\Temp\TL_5fa863ae-791f-4777-b72c-e4e1eb2880a8'... done!

Retrieving m3u8 playlist urls for all VOD qualities... done!
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/chunked/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/720p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/480p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/audio_only/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/360p30/index-dvr.m3u8
https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/160p30/index-dvr.m3u8

Playlist url for selected quality 1920x1080@30fps (Source) is https://d2nvs31859zcd8.cloudfront.net/333c43bdacdcfbf31731_a_couple_streams_41496491134_1611878332/chunked/index-dvr.m3u8

Retrieving playlist... done!
Parsing playlist... done!
Number of video chunks: 1213

Starting parallel video chunk download
Number of video chunks to download: 1213
Maximum connection count: 10

Parallel video chunk download is running...

Download of all video chunks complete!

Merging all VOD parts into 'D:\Videos\Captures\20210128_891965995_Music.ts'...

Starting temporary download folder cleanup!
Deleting directory 'C:\Users\haris\AppData\Local\Temp\TL_5fa863ae-791f-4777-b72c-e4e1eb2880a8'... done!

Download task ended with an error!

System.AggregateException: One or more errors occurred. ---> System.IO.FileNotFoundException: Could not find file 'D:\Videos\Captures\20210128_891965995_Music.ts'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
   at TwitchLeecher.Services.Services.ProcessingService.ConcatParts(Action`1 log, Action`1 setStatus, Action`1 setProgress, VodPlaylist vodPlaylist, String concatFile)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass51_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.IO.FileNotFoundException: Could not find file 'D:\Videos\Captures\20210128_891965995_Music.ts'.
File name: 'D:\Videos\Captures\20210128_891965995_Music.ts'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
   at TwitchLeecher.Services.Services.ProcessingService.ConcatParts(Action`1 log, Action`1 setStatus, Action`1 setProgress, VodPlaylist vodPlaylist, String concatFile)
   at TwitchLeecher.Services.Services.TwitchService.<>c__DisplayClass51_0.<StartQueuedDownloadIfExists>b__2()
   at System.Threading.Tasks.Task.Execute()<---

**Comment by Franiac** [comment]
What is the directory `D:\Videos\Captures\`?

`D:\Videos\Captures\20210128_891965995_Music.mp4: No such file or directory`

`Could not find file 'D:\Videos\Captures\20210128_891965995_Music.ts'`

Because TL seems to have a problem finding this directory. Is it an external drive or a NAS storage path?
Does the directory exist?
Does TL have write access to it?

I don't really see the problem to be honest. Usually TL creates non-existing directories.

**Comment by Harison117** [comment]
What is the directory D:\Videos\Captures\?
It'a folder that I created for storing the videos from TL

Because TL seems to have a problem finding this directory. Is it an external drive or a NAS storage path?
No It's my HD :(

Does TL have write access to it?
I think thats the issue, cause I just realized(trying to download some work files) that my PC is not allowing other programs to do anything with my computer(like download games from steam, or download something else using google chrome...)
I dont know what happened in this past days, but, I'll try to fix it and let u know if I find another problem.
Thanks for your help
